fsspec/implementations/tests/test_hdfs.py

```diff
@@ -61,3 +73,53 @@ def test_read(hdfs):
         assert f.read() == data
     with h.open(out, "rb") as f:
         assert f.read(100) + f.read() == data
+
+
+def test_copying(hdfs):
+    fs = fsspec.filesystem("hdfs")
+
+    fs.mkdir(basedir + "/test_dir")
+    fs.touch(basedir + "/test_dir/a")
+    fs.touch(basedir + "/test_dir/b")
+    fs.mkdir(basedir + "/test_dir/c")
+    fs.touch(basedir + "/test_dir/c/d")
+
+    fs.copy(basedir + "/test_dir", basedir + "/copy_dir", recursive=True)
+    assert fs.find(basedir + "/copy_dir", detail=False) == [
+        basedir + "/copy_dir" + "/a",
+        basedir + "/copy_dir" + "/b",
+        basedir + "/copy_dir" + "/c/d",
+    ]
+
+
+def test_getting(hdfs, tmpdir):
+    fs = fsspec.filesystem("hdfs")
+
+    src_dir = Path(tmpdir / "source")
+    dst_dir = Path(tmpdir / "destination")
+
+    src_dir.mkdir()
+    (src_dir / "file_1.txt").write_text("file_1")
+    (src_dir / "file_2.txt").write_text("file_2")
+    (src_dir / "dir_1").mkdir()
+    (src_dir / "dir_1" / "file_3.txt").write_text("file_3")
+    (src_dir / "dir_1" / "file_4.txt").write_text("file_4")
+    (src_dir / "dir_2").mkdir()
+
+    fs.put(str(src_dir), basedir + "/src", recursive=True)
+    fs.get(basedir + "/src", str(dst_dir), recursive=True)
+
+    files = [file.relative_to(dst_dir) for file in dst_dir.glob("**/*")]
+
+    assert set(map(str, files)) == {
+        "file_1.txt",
+        "file_2.txt",
+        "dir_1/file_3.txt",
+        "dir_1/file_4.txt",
+        "dir_1",
+        "dir_2",
+    }
```

Comment:
I think this test should cover the other new behavior we saw.

Reply:
Which one?

Reply:
If I'm reading the test correctly, this should cover the relative vs. absolute path behavior when copying directories, like the example shown in my most recent comment on #777 .